source/core/solid_base.cc

```diff
@@ -482,18 +486,21 @@ SolidBase<dim, spacedim>::integrate_velocity(double time_step)
   // than unity
   for (unsigned int it = 0; it < sub_particles_iterations; ++it)
     {
+      // Set the time of the velocity function
       for (auto particle = solid_particle_handler->begin();
            particle != solid_particle_handler->end();
            ++particle)
         {
           Point<spacedim> particle_location = particle->get_location();
 
           Tensor<1, spacedim> k1;
```

Comment:
I would indicate in a comment what these are (particles velocity at different time step?) + maybe use a more explicit variable name (such as `v1`, for velocity)

Reply:
I'm not sure I understand your comment. This function integrates the particle velocity, hence it's pretty clear that it is the particle velocities no?
I can remove line 489. It's confusing.

Reply:
I think the key misunderstanding here is that I missed it is a classical RK4 scheme. I would put this comment at the beginning of this section (like, at line 489):

`// A classic Runge-Kutta scheme (RK4) is used for the particles velocities. See https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods for more details.`

Then at line 495, just before `k1` definition:

`// RK4 coefficients: k1, k2, k3, k4`